test/unit-test/Gen2GetTests.cpp

```diff
@@ -133,6 +132,86 @@ BOOST_AUTO_TEST_CASE(Gen2_Get_Sensor) {
   BOOST_TEST(res == expectedJson);
 }
 
+BOOST_AUTO_TEST_CASE(Gen2_Get_Invalid_JSON) {
+  WsChannel channel;
+  channel.setAuthorized(false);
+  channel.setConnID(1);
+
+  jsoncons::json jsonSetRequestForSignal;
+
+  std::string path{"Vehicle/Speed"};
+  const VSSPath vss_path = VSSPath::fromVSSGen2(path);
+
+  jsonSetRequestForSignal["action"] = "get";
+  jsonSetRequestForSignal["path"] = path;
+  jsonSetRequestForSignal["requestId"] = 100;  // int is invalid here;
+
+  std::string expectedJsonString{R"(
+      {
+  "action": "get",
+  "error": {
+    "message": "Schema error: VSS get malformed: #/requestId: Expected string, found uint64",
+    "number": 400,
+    "reason": "Bad Request"
+  },
+  "requestId": "100",
+  "timestamp": 0
+}
+      )"};
+  jsoncons::json expectedJson = jsoncons::json::parse(expectedJsonString);
+
+  // run UUT
+  auto resStr =
+      processor->processQuery(jsonSetRequestForSignal.as_string(), channel);
+  auto res = json::parse(resStr);
+
+  // Does result have a timestamp?
+  BOOST_TEST(res["timestamp"].as<int64_t>() > 0);
+
+  // Remove timestamp for comparision purposes
+  expectedJson["timestamp"] = res["timestamp"].as<int64_t>();
+
+  BOOST_TEST(res == expectedJson);
+}
+
+/** Send an invalid JSON, without any determinable Request Id */
+BOOST_AUTO_TEST_CASE(Gen2_Get_Invalid_JSON_NoRequestID) {
+  WsChannel channel;
+  channel.setAuthorized(false);
+  channel.setConnID(1);
+
+  jsoncons::json jsonSetRequestForSignal;
+
+  jsonSetRequestForSignal["action"] = "get";
+  jsonSetRequestForSignal["path"] = 999;  // int as path is wrong
+
+  std::string expectedJsonString{R"(
+  {
+  "action": "get",
+  "error": {
+    "message": "Schema error: VSS get malformed: #: Required property \"requestId\" not found\n#/path: Expected string, found uint64",
+    "number": 400,
+    "reason": "Bad Request"
+  },
+  "requestId": "UNKNOWN",
+  "timestamp": 0
+}
+      )"};
+  jsoncons::json expectedJson = jsoncons::json::parse(expectedJsonString);
+
+  // run UUT
```

Comment:
what is UUT?

Reply:
I took the style form the existing tests, i assume "Unit under test"